### PR TITLE
daemon,cmd/snapd: allow daemon.New() to fail

### PR DESCRIPTION
**Pull request by zyga**
This patch changes daemon.New() to return the created Daemon instance as
well as an error. This lets us get rid of panic() calls that started to
swarm in there.

Signed-off-by: Zygmunt Krynicki <zygmunt.krynicki@canonical.com>

### CODE REVIEW

File: daemon/daemon_test.go [hunk 1]
```diff
@@ -36,8 +36,10 @@ type daemonSuite struct{}
 var _ = check.Suite(&daemonSuite{})
 
 // build a new daemon, with only a little of Init(), suitable for the tests
-func newTestDaemon() *Daemon {
-	d := New()
+func newTestDaemon(c *check.C) *Daemon {
+	d, err := New()
+	c.Assert(err, check.IsNil)
+	c.Assert(d, check.Not(check.IsNil))
```
Comment (stevenwilkin) [inline daemon/daemon_test.go]: This is maybe a bit redundant after asserting that the error is nil.

**Comment by stevenwilkin** [comment]
:+1:

**Comment by zyga** [comment]
retest this please

File: daemon/daemon.go [hunk 2]
```diff
@@ -265,19 +265,19 @@ func getTrustedAccountKey() string {
...
 	db, err := asserts.OpenSysDatabase(getTrustedAccountKey())
 	if err != nil {
-		panic(err.Error())
+		return nil, err
 	}
 	interfacesRepo := interfaces.NewRepository()
 	for _, iface := range builtin.Interfaces() {
 		if err := interfacesRepo.AddInterface(iface); err != nil {
-			panic(err.Error())
+			return nil, err
```
Comment (chipaca) [inline daemon/daemon.go]: isn't there cleanup to be done here? Of the `*Database` and `*Overlord` you now have I mean.

Comment (chipaca) [inline daemon/daemon.go]: pedronis says "not really". Still makes me nervous... oh well.

File: daemon/daemon_test.go (hunk 1 above)
Comment (chipaca) [inline daemon/daemon_test.go]: I'd usually write that `check.NotNil` " :-)

Comment (niemeyer) [inline daemon/daemon_test.go]: Agree with @stevenwilkin, we don't check for the value not-nillness elsewhere after we've tested the err.

**Comment by niemeyer** [comment]
LGTM

**Pull request merged**